src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -892,11 +877,10 @@ private static List<VBuffer<ReadOnlyMemory<char>>> Train(IHostEnvironment env, I
             // Initialize all LDA states
             for (int i = 0; i < columns.Length; i++)
             {
-                var state = new LdaState(env, columns[i], numVocabs[i]);
-
                 if (numDocArray[i] == 0 || corpusSize[i] == 0)
                     throw ch.Except("The specified documents are all empty in column '{0}'.", columns[i].InputColumnName);
 
+                var state = new LdaState(env, columns[i], numVocabs[i]);
```

Comment:
📝 This is a fairly straightforward change. We avoid creating an instance of `LdaState` which would result in an exception during dispose, for the case where an exception is thrown two lines above.

Reply:
This change is no longer necessary after #4551